src/cascadia/TerminalSettingsModel/GlobalAppSettings.h

```diff
@@ -52,43 +54,49 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
-        GETSET_PROPERTY(bool, SnapToGridOnResize, true);
-        GETSET_PROPERTY(bool, ForceFullRepaintRendering, false);
-        GETSET_PROPERTY(bool, SoftwareRendering, false);
-        GETSET_PROPERTY(bool, ForceVTInput, false);
-        GETSET_PROPERTY(bool, DebugFeaturesEnabled); // default value set in constructor
-        GETSET_PROPERTY(bool, StartOnUserLogin, false);
-        GETSET_PROPERTY(bool, AlwaysOnTop, false);
-        GETSET_PROPERTY(bool, UseTabSwitcher, true);
-        GETSET_PROPERTY(bool, DisableAnimations, false);
+        bool HasUnparsedDefaultProfile() const;
+        winrt::hstring UnparsedDefaultProfile() const;
+        void UnparsedDefaultProfile(const hstring& value);
+        void ClearUnparsedDefaultProfile();
+
+        GETSET_SETTING(int32_t, InitialRows, DEFAULT_ROWS);
+        GETSET_SETTING(int32_t, InitialCols, DEFAULT_COLS);
+        GETSET_SETTING(bool, AlwaysShowTabs, true);
+        GETSET_SETTING(bool, ShowTitleInTitlebar, true);
+        GETSET_SETTING(bool, ConfirmCloseAllTabs, true);
+        GETSET_SETTING(winrt::Windows::UI::Xaml::ElementTheme, Theme, winrt::Windows::UI::Xaml::ElementTheme::Default);
+        GETSET_SETTING(winrt::Microsoft::UI::Xaml::Controls::TabViewWidthMode, TabWidthMode, winrt::Microsoft::UI::Xaml::Controls::TabViewWidthMode::Equal);
+        GETSET_SETTING(bool, ShowTabsInTitlebar, true);
+        GETSET_SETTING(hstring, WordDelimiters, DEFAULT_WORD_DELIMITERS);
+        GETSET_SETTING(bool, CopyOnSelect, false);
+        GETSET_SETTING(winrt::Microsoft::Terminal::TerminalControl::CopyFormat, CopyFormatting, 0);
+        GETSET_SETTING(bool, WarnAboutLargePaste, true);
+        GETSET_SETTING(bool, WarnAboutMultiLinePaste, true);
+        GETSET_SETTING(Model::LaunchPosition, InitialPosition, nullptr, nullptr);
+        GETSET_SETTING(Model::LaunchMode, LaunchMode, LaunchMode::DefaultMode);
+        GETSET_SETTING(bool, SnapToGridOnResize, true);
+        GETSET_SETTING(bool, ForceFullRepaintRendering, false);
+        GETSET_SETTING(bool, SoftwareRendering, false);
+        GETSET_SETTING(bool, ForceVTInput, false);
+        GETSET_SETTING(bool, DebugFeaturesEnabled); // default value set in constructor
+        GETSET_SETTING(bool, StartOnUserLogin, false);
+        GETSET_SETTING(bool, AlwaysOnTop, false);
+        GETSET_SETTING(bool, UseTabSwitcher, true);
+        GETSET_SETTING(bool, DisableAnimations, false);
 
     private:
-        hstring _unparsedDefaultProfile;
         guid _defaultProfile;
+        std::optional<hstring> _UnparsedDefaultProfile{ std::nullopt };
+        bool _validDefaultProfile;
 
         com_ptr<KeyMapping> _keymap;
         std::vector<SettingsLoadWarnings> _keybindingsWarnings;
 
         Windows::Foundation::Collections::IMap<hstring, Model::ColorScheme> _colorSchemes;
         Windows::Foundation::Collections::IMap<hstring, Model::Command> _commands;
 
+        std::optional<hstring> _getUnparsedDefaultProfileImpl() const;

```

Comment:
how is this different from `GETSET_SETTING`?

Reply:
The setter for UnparsedDefaultProfile needs to set `_validDefaultProfile` to false. We need this to validate that the provided "defaultProfile" value actually maps to something.